runtime/src/bank_forks.rs

```diff
@@ -186,12 +204,12 @@ impl BankForks {
         }
     }
 
-    pub fn insert(&mut self, mut bank: Bank) -> Arc<Bank> {
+    pub fn insert(&mut self, mut bank: Bank) -> BankWithScheduler {
         bank.check_program_modification_slot =
             self.root.load(Ordering::Relaxed) < self.highest_slot_at_startup;
 
-        let bank = Arc::new(bank);
-        let prev = self.banks.insert(bank.slot(), bank.clone());
+        let bank = BankWithScheduler::new_without_scheduler(Arc::new(bank));
```

Comment:
this is the sole call site of `BankWithScheduler` constructor, and it is always creating with no scheduler for now.

Reply:
It looks like we call it above for the insertion of initial forks as well? in `new_from_banks`

Might just be misunderstanding; this is the only call site where we plan to construct with a scheduler in the future, right?

Reply:
> this is the only call site where we plan to construct **with a scheduler** (emphasis mine) in the future, right?

yeah. `new_from_banks` shouldn't have scheduler because they should be frozen, iirc.